service/worker/deletenamespace/deleteexecutions/activities.go

```diff
@@ -126,10 +127,29 @@ func (a *LocalActivities) GetNextPageTokenActivity(ctx context.Context, params G
 func (a *Activities) DeleteExecutionsActivity(ctx context.Context, params DeleteExecutionsActivityParams) (DeleteExecutionsActivityResult, error) {
 	ctx = headers.SetCallerName(ctx, params.Namespace.String())
 
-	rateLimiter := quotas.NewRateLimiter(float64(params.RPS), params.RPS)
+	progressCh := make(chan DeleteExecutionsActivityResult, 1)
+	defer func() { close(progressCh) }()
+
+	go func() {
+		heartbeatTicker := time.NewTicker(deleteWorkflowExecutionsActivityOptions.HeartbeatTimeout / 2)
+		defer heartbeatTicker.Stop()
+
+		var lastKnownProgress DeleteExecutionsActivityResult
```

Comment:
This should be initialized to the previous heartbeat details to avoid losing progress.

Reply:
This is basically statistics and tool to monitor the progress in UI. It won't be 100% accurate and nothing really depends on it. But, I agree, `SuccessCount` should be carried over to be more accurate.